src/mscorlib/src/System/Globalization/CompareInfo.cs

```diff
@@ -459,6 +491,34 @@ internal static unsafe int CompareOrdinalIgnoreCase(string strA, int indexA, int
                     length--;
                 }
 
+                if (CultureData.InvariantMode)
```

Comment:
is it necessary to duplicate this block? can't the condition above be changed from
`while (length != 0 && (*a <= 0x80) && (*b <= 0x80))`
to
`while (length != 0 && ((*a <= 0x80) && (*b <= 0x80)) || CultureData.InvariantMode)`